spec/dummy_models_total_count_controller_spec.rb

```diff
@@ -0,0 +1,144 @@
+require 'spec_helper'
+
+describe DummyModelsTotalCountController, type: :controller do
+  describe '#index' do
+    let!(:dummy_models) { create_list(:dummy_model, 28) }
+    let(:expected_list) do
+      dummy_models.first(25).map do |dummy|
+        { 'id' => dummy.id, 'name' => dummy.name, 'something' => dummy.something }
+      end
+    end
+
+    context 'when paginating an ActiveRecord with no previous pagination but kaminari installed' do
+      before do
+        get :index
+      end
+
+      include_examples 'total count pagination param'
+
+      it 'responds with valid page' do
+        expect(response_body(response)['page']).to eq expected_list
+      end
+    end
+
+    context 'when paginating with page and limit params' do
+      context 'with a particular limit passed by option' do
+        let(:expected_list) do
+          dummy = dummy_models.third
+          [{ 'id' => dummy.id, 'name' => dummy.name, 'something' => dummy.something }]
+        end
+
+        before do
+          get :index_with_params
```

Comment:
```suggestion
          get :index_with_params, params: { page: page, limit: limit }
```

Reply:
I don't think this would be necessary, the endpoint `index_with_params` includes the params inside the controller method